Rollbar.NetPlatformExtensions/RollbarLogger.cs

```diff
@@ -203,52 +202,44 @@ mslogging.EventId eventId
 
         private bool disposedValue = false; // To detect redundant calls
 
+
         /// <summary>
         /// Releases unmanaged and - optionally - managed resources.
         /// </summary>
         /// <param name="disposing">
         ///   <c>true</c> to release both managed and unmanaged resources; 
         ///   <c>false</c> to release only unmanaged resources.
         /// </param>
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Major Code Smell", "S1066:Collapsible \"if\" statements should be merged", Justification = "Cleaner code structure.")]
         protected virtual void Dispose(bool disposing)
         {
             if (!disposedValue)
             {
                 if (disposing)
                 {
-                    // TODO: dispose managed state (managed objects).
+                    // dispose managed state (managed objects).
 
                     if (this._rollbar != RollbarLocator.RollbarInstance)
                     {
                         this._rollbar.Dispose();
                     }
                 }
 
-                // TODO: free unmanaged resources (unmanaged objects) and override a finalizer below.
-                // TODO: set large fields to null.
+                // free unmanaged resources (unmanaged objects) and override a finalizer below.
+                // set large fields to null.
 
                 disposedValue = true;
             }
         }
 
-
-        // TODO: override a finalizer only if Dispose(bool disposing) above has code to free unmanaged resources.
-        // ~RollbarLogger() {
-        //   // Do not change this code. Put cleanup code in Dispose(bool disposing) above.
-        //   Dispose(false);
-        // }
-
         /// <summary>
         /// Performs application-defined tasks associated with freeing, releasing, or resetting unmanaged resources.
         /// </summary>
-#pragma warning disable CA1063 // Implement IDisposable Correctly
         public void Dispose()
-#pragma warning restore CA1063 // Implement IDisposable Correctly
         {
             // Do not change this code. Put cleanup code in Dispose(bool disposing) above.
             Dispose(true);
-            // TODO: uncomment the following line if the finalizer is overridden above.
-            // GC.SuppressFinalize(this);
+            GC.SuppressFinalize(this);
```

Comment:
I didn't see any code change in `Dispose(bool disposing)`. Is the finalizer overridden as described in the comment? Or was the comment incorrect, or stale?

Reply:
It appears that now SuppressFinalize(this) can be called in either case. That would make the legacy code template that had it commented out by default incorrect.